common/remote/hooks/azure/azurefile.cpp

```diff
@@ -1032,48 +1040,45 @@ class CAzureApiCopyClient : public CInterfaceOf<IAPICopyClient>
         }
         return false;
     }
-    virtual IAPICopyClientOp * startCopy(const char *srcPath, unsigned srcStripeNum,  const char *tgtPath, unsigned tgtStripeNum) override
+    virtual IAPICopyClientOp * startCopy(const char *srcPath, unsigned srcStripeNum,  const char *tgtPath, unsigned tgtStripeNum) const override
     {
-        if (!haveSASTokens)
-        {
-            sourceApiInfo->getSASToken(sourceSASToken);
-            targetApiInfo->getSASToken(targetSASToken);
-            haveSASTokens=true;
-        }
-        StringBuffer sourceURI, targetURI;
-        buildURI(sourceURI, sourceURIbase, srcStripeNum,  srcPath, sourceApiInfo, sourceSASToken);
-        buildURI(targetURI, targetURIbase, tgtStripeNum,  tgtPath, targetApiInfo, targetSASToken);
+        CAzureURI targetURI(tgtStripeNum,  tgtPath, targetApiInfo);
         Owned<IAPICopyClientOp> apiClient;
         if (isAzureFile(targetApiInfo->getStorageType()))
-            apiClient.setown(new AzureFileClient());
+            apiClient.setown(new AzureFileClient(targetURI.str()));
         else
-            apiClient.setown(new AzureBlobClient());
-        apiClient->startCopy(sourceURI, targetURI);
+            apiClient.setown(new AzureBlobClient(targetURI.str()));
+
+        CAzureURI sourceURI(srcStripeNum,  srcPath, sourceApiInfo);
+        apiClient->startCopy(sourceURI.str());
         return apiClient.getClear();
     }
 protected:
-    void buildStorageURIBase(IStorageApiInfo * storageApiInfo, StringBuffer & uri)
+    class CAzureURI
     {
-        const char *accountName = storageApiInfo->queryStorageApiAccount();
-        uri.appendf("https://%s", accountName);
-
-        if (isAzureFile(targetApiInfo->getStorageType()))
-        {
-            uri.append(".file");
-        }
-        else
-        {
-            dbgassertex(isAzureBlob(targetApiInfo->getStorageType()));
-            uri.append(".blob");
-        }
-        uri.append(".core.windows.net/");
-    }
-    void buildURI(StringBuffer & fulluri, const char * baseURI, unsigned stripeNum, const char *filePath, IStorageApiInfo *apiInfo, const char *token)
-    {
-        StringBuffer tmp;
-        const char * container = apiInfo->queryStorageContainer(stripeNum);
-        fulluri.appendf("%s%s%s%s", baseURI, container, encodeURL(tmp, filePath).str(), token);
-    }
+        public:
+            CAzureURI(unsigned stripeNum, const char *filePath, const IStorageApiInfo *apiInfo)
```

Comment:
> CAzureURI sourceURI(srcStripeNum,  srcPath, sourceApiInfo);
> apiClient->startCopy(sourceURI.str());

personal: a class/object seems to be a bit overkill/less clear than a helper function (unless the intention is to add extra functionality to this class that does more). e.g.:
```
StringBuffer srcUri;
getAzureURI(srcUri, srcPath, srcStripeNum,  srcPath, sourceApiInfo);
apiClient->startCopy(sourceURI.str());
```